apps/prairielearn/src/components/SubmissionPanel.html.ts

```diff
@@ -0,0 +1,549 @@
+                    </tr>
+                    <tr>
+                      <th><span class="text-success mr-2">&bull;</span>Run duration</th>
+                      <td>${gradingJobStats.runDuration}</td>
+                    </tr>
+                    <tr>
+                      <th><span class="text-danger mr-2">&bull;</span>Report duration</th>
+                      <td>${gradingJobStats.reportDuration}</td>
+                    </tr>
+                    <tr>
+                      <th>Total duration</th>
+                      <td>${gradingJobStats.totalDuration}</td>
+                    </tr>
+                  `
+                : ''}
+            </tbody>
+          </table>
+          ${question.grading_method === 'External'
+            ? html`
+                <div class="d-flex mt-2 mb-2">
+                  <span
+                    style="display: inline-block; width: ${gradingJobStats
+                      .phases[0]}%; height: 10px;"
+                    class="bg-dark m-0"
+                  ></span>
+                  <span
+                    style="display: inline-block; width: ${gradingJobStats
+                      .phases[1]}%; height: 10px;"
+                    class="bg-warning m-0"
+                  ></span>
+                  <span
+                    style="display: inline-block; width: ${gradingJobStats
+                      .phases[2]}%; height: 10px;"
+                    class="bg-primary m-0"
+                  ></span>
+                  <span
+                    style="display: inline-block; width: ${gradingJobStats
+                      .phases[3]}%; height: 10px;"
+                    class="bg-success m-0"
+                  ></span>
+                  <span
+                    style="display: inline-block; width: ${gradingJobStats
+                      .phases[4]}%; height: 10px;"
+                    class="bg-danger m-0"
+                  ></span>
+                </div>
+                ${course_instance_id != null
+                  ? html`
+                      <a
+                        class="btn btn-primary mt-2"
+                        href="${config.urlPrefix}/course_instance/${course_instance_id}/instructor/grading_job/${submission
+                          .grading_job?.id}"
+                        >View grading job ${submission.grading_job?.id}</a
+                      >
+                    `
+                  : html`
+                      <a
+                        class="btn btn-primary mt-2"
+                        href="${urlPrefix}/grading_job/${submission.grading_job?.id}"
+                        >View grading job ${submission.grading_job?.id}</a
+                      >
+                    `}
+              `
+            : ''}
+        `,
+    footer: html`
+      <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
+    `,
+  });
+}
+
+function buildGradingJobStats(job: GradingJob | null) {
+  if (job) {
```

Comment:
Nit: consider an early return `if (!job)` to reduce the indentation of the rest of the function.